File: backend/pet-owner-service/src/main/java/com/txakurrapp/petownerservice/model/Message.java
package com.txakurrapp.petownerservice.model;

import javax.persistence.*;
import java.time.LocalDateTime;

@Entity
public class Message {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private Long senderId;
    private Long recipientId;
    private LocalDateTime timeStamp;
    private String messageSubject;
    @Column(columnDefinition = "TEXT")
    private String messageBody;

    public Message() {
    }

    public Message(Long senderId, Long recipientId, String messageSubject, String messageBody) {
        this.senderId = senderId;
        this.recipientId = recipientId;
        this.timeStamp = LocalDateTime.now();
        this.messageSubject = messageSubject;
        this.messageBody = messageBody;
    }

//  Getters and setters
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getSenderId() {
        return senderId;
    }

    public void setSenderId(Long senderId) {
        this.senderId = senderId;
    }

    public Long getRecipientId() {
        return recipientId;
    }

    public void setRecipientId(Long recipientId) {
        this.recipientId = recipientId;
    }

    public LocalDateTime getTimeStamp() {
        return timeStamp;
    }

    public void setTimeStamp(LocalDateTime timeStamp) {
        this.timeStamp = timeStamp;
    }

    public String getMessageSubject() {
        return messageSubject;
    }

    public void setMessageSubject(String messageSubject) {
        this.messageSubject = messageSubject;
    }

    public String getMessageBody() {
        return messageBody;
    }

    public void setMessageBody(String messageBody) {
        this.messageBody = messageBody;
    }
}

File: frontend/txakurrapp/src/app/classes/petowner/message.ts
// Matches MessageGetDTO in Edge Service
export class Message {
    
    constructor(
        private _id: number,
        private _senderId: number,
        private _recipientId: number,
        private _timeStamp: Date,
        private _messageSubject: string,
        private _messageBody: string,
    ){}
    
    public get id(): number {
        return this._id;
    }
    public set id(value: number) {
        this._id = value;
    }
    public get senderId(): number {
        return this._senderId;
    }
    public set senderId(value: number) {
        this._senderId = value;
    }
    public get recipientId(): number {
        return this._recipientId;
    }
    public set recipientId(value: number) {
        this._recipientId = value;
    }
    public get timeStamp(): Date {
        return this._timeStamp;
    }
    public set timeStamp(value: Date) {
        this._timeStamp = value;
    }
    public get messageSubject(): string {
        return this._messageSubject;
    }
    public set messageSubject(value: string) {
        this._messageSubject = value;
    }
    public get messageBody(): string {
        return this._messageBody;
    }
    public set messageBody(value: string) {
        this._messageBody = value;
    }

}
